src/c_api/c_api_error.h

```diff
@@ -52,4 +52,12 @@ inline int XGBAPIHandleException(const dmlc::Error &e) {
   XGBAPISetLastError(e.what());
   return -1;
 }
+
+#define xgboost_CHECK_C_ARG_PTR(out_ptr)                      \
+  do {                                                        \
```

Comment:
Sorry, came across this code randomly, why is it wrapped with a `do { } while (0)` ?